lib/game/base_game.dart

```diff
@@ -129,9 +129,10 @@ class BaseGame extends Game with FPSCounter {
     });
     _removeLater.clear();
 
-    components.addAll(_addLater);
-    _addLater.forEach((component) => component.onMount());
+    final addNow = _addLater.toList(growable: false);
```

Comment:
I have one doubt, and one suggestion about this.

Doubt: Will this add any perfomance overhead? If this do, we could check the _addLater length, and only copy the list if length > 0

Suggestion `addNow` and `addLater` got a little confusing here? Maybe `toAddComponents` or something like that. Maybe I am the problem so feel free to ignore this suggestion.

Reply:
A list of length 0 should be extremely cheap to copy, I think it is better to have it like this for readability.

On the topic of readability, you are correct that the naming might confusing, not sure if `toAddComponents` are less confusing though? Let's see what the other reviewers think.

Reply:
> A list of length 0 should be extremely cheap to copy, I think it is better to have it like this for readability.
Not sure about that, on a game engine I would like to squeeze every bit of perfomance possible, and I don't think wrapping this small section of code in an `if` would impact that much on the perfomance.

 > not sure if toAddComponents is less confusing though?
I think it is less confusing, but I didn't liked it very much, but couldn't think on anything better 😅 hahaha

Reply:
You're right, it actually got better when wrapping it. :) 
I thought it was just going to wrap a line, but it could wrap the whole thing.